Model/Data/CollectData.php

```diff
@@ -0,0 +1,121 @@
+    private $cmsFeaturesDataHelper;
+    /**
+     * @var Request
+     */
+    private $request;
+    /**
+     * @var Logger
+     */
+    private $logger;
+
+    /**
+     * @var ApiConfigHelper
+     */
+    private $apiConfigHelper;
+
+
+    const HEADER_SIGNATURE_KEY = 'X-Alma-Signature';
+    /**
+     * @var Config
+     */
+    private $config;
+
+
+    /**
+     * @param Logger $logger
+     * @param PayloadFormatter $payloadFormatter
+     * @param CmsInfoDataHelper $cmsInfoDataHelper
+     * @param CmsFeaturesDataHelper $cmsFeaturesDataHelper
+     * @param Request $request
+     * @param ApiConfigHelper $apiConfigHelper
+     * @param Config $config
+     */
+    public function __construct(
+        Logger                $logger,
+        PayloadFormatter      $payloadFormatter,
+        CmsInfoDataHelper     $cmsInfoDataHelper,
+        CmsFeaturesDataHelper $cmsFeaturesDataHelper,
+        Request               $request,
+        ApiConfigHelper       $apiConfigHelper,
+        Config                $config
+    )
+    {
+        $this->payloadFormatter = $payloadFormatter;
+        $this->cmsInfoDataHelper = $cmsInfoDataHelper;
+        $this->cmsFeaturesDataHelper = $cmsFeaturesDataHelper;
+        $this->request = $request;
+        $this->logger = $logger;
+        $this->apiConfigHelper = $apiConfigHelper;
+        $this->config = $config;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function collect()
+    {
+        try {
+            $this->checkSignature($this->config->getMerchantId(), $this->getApiKey());
+        } catch (CollectDataException $e) {
+            throw new Exception(__($e->getMessage()), 0, Exception::HTTP_FORBIDDEN);
+        }
+        $cmsInfo = new CmsInfo($this->cmsInfoDataHelper->getCmsInfoData());
+        $cmsFeatures = new CmsFeatures($this->cmsFeaturesDataHelper->getCmsFeaturesData());
+
+        return $this->payloadFormatter->formatConfigurationPayload($cmsInfo, $cmsFeatures);
+    }
+
+    /**
+     * @param string | null $merchant_id
+     * @param string $apiKey
+     * @return void
+     * @throws CollectDataException
+     */
+    private function checkSignature(?string $merchant_id, string $apiKey): void
+    {
+        $signature = $this->request->getHeader(self::HEADER_SIGNATURE_KEY);
+        if (!$signature) {
+            throw new CollectDataException("Missing signature");
+        }
+
+        if (!RequestUtils::isHmacValidated($merchant_id, $apiKey, $signature)) {
+            $this->logger->error("Wrong signature in collect request", [
+                '$merchant_id' => $merchant_id,
+                'signature' => $signature
+            ]);
+            throw new CollectDataException("Wrong signature in collect request");
+        }
+    }
+
+    private function getApiKey(): string
+    {
+        $apiKey = $this->apiConfigHelper->getActiveAPIKey();
+        if (!$apiKey) {
+            throw new CollectDataException("Missing API key in collect request");
```

Comment:
Isn't better to use this throw directly in the `getActiveAPIKey` method and get directly this method instead of `getApiKey` ?